modules/postgres-instance/basic/main.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  # Name restrictions https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
+  default_name = trimsuffix(lower(substr("${var.prefix}${var.app_id}-${var.env_id}-${replace(var.res_id, ".", "-")}", 0, 63)), "-")
+
+  name = coalesce(var.name, local.default_name)
+
+  default_username = "main"
+
+  tags = {
+    "managed-by" = "terraform"
+    "humanitec"  = "true"
+    "hum-app"    = var.app_id
+    "hum-env"    = var.env_id
+    "hum-res"    = var.res_id
+  }
+}
+
+data "azurerm_resource_group" "postgres" {
+  name = var.resource_group_name
+}
+
+data "azurerm_resource_group" "workload" {
+  name = var.workload_resource_group_name
+}
+
+data "azurerm_virtual_network" "vn" {
+  name                = var.virtual_network_name
```

Comment:
do we need ingress/firewall rules to allow access to the private endpoint?

Reply:
No, they seem to be reachable by default.